### PR TITLE
refact: AuthZ write init, path and casbin layer 

### PR DESCRIPTION
**Pull request by moogacs**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/auth/authorization/rbac/manager.go [hunk 1]
```diff
@@ -40,93 +51,135 @@ func (m *Manager) AddPolicies(policies []*Policy) error {
...
+	for _, role := range roles {
 		if _, err := m.casbin.DeleteRoleForUser(user, role); err != nil {
 			return err
 		}
 	}
 	if err := m.casbin.SavePolicy(); err != nil {
 		return err
 	}
-	return nil
+	return m.casbin.InvalidateCache()
```
Comment (dirkkul) [inline usecases/auth/authorization/rbac/manager.go:147]: general question: When do we need to to invalidate this and when do we not? Only after deletes?

File: usecases/auth/authorization/rbac/types..go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	// userPrefix = "u_"
+)
+
+var builtInRoles = map[string]string{
+	"viewer": authorization.READ,
+	"editor": authorization.CRU,
+	"admin":  authorization.CRUD,
+}
+
+func policy(permission *models.Permission) (resource, verb, domain string) {
```
Comment (dirkkul) [inline usecases/auth/authorization/rbac/types..go:40]: can we keep domain as a custom type like it was before?

File: usecases/auth/authorization/rbac/types..go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		resource = authorization.Collections(*permission.Collection)[0]
+	case tenants:
+		resource = authorization.Shards(*permission.Collection, *permission.Tenant)[0]
+	case objects:
+		resource = authorization.Objects(*permission.Collection, *permission.Tenant, strfmt.UUID(*permission.Object))
+	}
+	return
+}
+
+func permission(policy []string) *models.Permission {
```
Comment (dirkkul) [inline usecases/auth/authorization/rbac/types..go:64]: let's have explizit parameters and not do multiple paramter in a list

**Review by dirkkul — CHANGES_REQUESTED** [review]

File: test/acceptance/authz/roles_test.go [hunk 4]
```diff
@@ -53,16 +54,17 @@ func TestAuthzRoles(t *testing.T) {
 	t.Run("get all roles before create", func(t *testing.T) {
 		res, err := helper.Client(t).Authz.GetRoles(authz.NewGetRolesParams(), clientAuth)
 		require.Nil(t, err)
-		require.Equal(t, 1, len(res.Payload))
-		require.Equal(t, existingRole, *res.Payload[0].Name)
+		require.Equal(t, 3, len(res.Payload))
+		// require.Equal(t, existingRole, *res.Payload[0].Name)
```
[the remark this answers is not in the document]
Comment (moogacs) [reply test/acceptance/authz/roles_test.go:58]: we have default 3 roles , need different pattern to check the names

File: usecases/auth/authorization/authorizer_test.go [hunk 5]
```diff
@@ -11,47 +11,36 @@
...
+// 				},
+// 			},
+// 		}
+
+// 		authorizer := adminlist.New(cfg.Authorization.AdminList)
+
+// 		_, ok := authorizer.(*adminlist.Authorizer)
+// 		assert.Equal(t, true, ok)
+// 	})
+// }
```
[the remark this answers is not in the document]
Comment (moogacs) [reply usecases/auth/authorization/authorizer_test.go:46]: it's not relevant anymore

File: usecases/auth/authorization/rbac/manager.go [hunk 6]
```diff
@@ -40,93 +51,135 @@ func (m *Manager) AddPolicies(policies []*Policy) error {
...
-	}
-	return nil
+	return m.casbin.InvalidateCache()
 }
 
-func (m *Manager) AddRolesForUser(user string, roles []string) error {
+func (m *manager) AddRolesForUser(user string, roles []string) error {
+	// userName := fmt.Sprintf("%s%s", userPrefix, user)
 	for _, role := range roles {
+		// roleName := fmt.Sprintf("%s%s", rolePrefix, role)
```
[the remark this answers is not in the document]
Comment (moogacs) [reply usecases/auth/authorization/rbac/manager.go:116]: that is not debug. that is gonna come next later on to prefix roles and users, but didn't wanna add it to this patch

File: usecases/auth/authorization/rbac/manager.go [hunk 7]
```diff
@@ -40,93 +51,135 @@ func (m *Manager) AddPolicies(policies []*Policy) error {
...
-	return nil
+	return m.casbin.InvalidateCache()
 }
 
-func (m *Manager) DeleteRoleFromUsers(role string) error {
-	users, err := m.casbin.GetUsersForRole(role)
-	if err != nil {
-		return err
+// Authorize will give full access (to any resource!) if the user is part of
+// the admin list or no access at all if they are not
```
[the remark this answers is not in the document]
Comment (moogacs) [reply usecases/auth/authorization/rbac/manager.go:151]: good catch

File: usecases/auth/authorization/rbac/manager.go (hunk 1 above)
Comment (moogacs) [reply usecases/auth/authorization/rbac/manager.go:147]: deletes or adds any changes in the roles

File: usecases/auth/authorization/types.go [hunk 8]
```diff
@@ -32,6 +32,15 @@ const (
...
 )
 
+var Actions = map[string]string{
+	CRUD:   "manage",
+	CRU:    "view",
+	CREATE: "create",
+	READ:   "read",
+	UPDATE: "update",
+	DELETE: "delete",
+}
```
[the remark this answers is not in the document]
Comment (moogacs) [reply usecases/auth/authorization/types.go:42]: again, that will affect multiple areas in the code, didn't wanna have that in this change

**Pull request merged**